### ISSUE TITLE
False positives in Python packages scans

### ISSUE THREAD

**Issue by jpopelka** [dependency-check/DependencyCheck#694]
First of all, let me thank you for outstanding tool. The Java analyzer works wery well.

Recently I've started to use the experimental Python analyzer and found out that there are many false positives. Many scans of python packages contain CVEs which are actually CVEs in Python interpreter itself.

For example [requests:2.13.0](https://pypi.python.org/pypi/requests) scan results contain:

```
           <identifiers>
                <identifier type="cpe" confidence="LOW">
                    <name>(cpe:/a:python-requests:requests:2.13.0)</name>
                </identifier>
                <identifier type="cpe" confidence="LOW">
                    <name>(cpe:/a:python:python:2.13.0)</name>
                </identifier>
                <identifier type="cpe" confidence="LOW">
                    <name>(cpe:/a:python:requests:2.13.0)</name>
                </identifier>
            </identifiers>
```

where I guess the `cpe:/a:python:python:2.13.0` does all the mess.

Do you think this can be fixed somehow or should I consider using the [Suppressing False Positives](http://jeremylong.github.io/DependencyCheck/general/suppression.html) ? Or any other idea how to avoid those false positives in Python packages scans ?

Thank you in advance.

**Comment by jeremylong** [comment]
Thanks for reporting the bug. I've been heads down in trying to update/resolve some of the core engine. Yes, this will likely be handled via the suppression XML.

**Comment by danawaze** [comment]
FYI i believe this is happening on python-dateutil as well - we're  using python 3.8 and have a dependency on python-dateutil 2.8.2 (was same false positive on 2.8.1, btw):

we get vulnerabilities summarised:

Dependency | Vulnerability IDs | Package | Highest Severity | CVE Count | Confidence | Evidence Count
-- | -- | -- | -- | -- | -- | --
python-dateutil:2.8.2 | `cpe:2.3:a:python:python:2.8.2:*:*:*:*:*:*:*`  `cpe:2.3:a:python_software_foundation:python:2.8.2:*:*:*:*:*:*:*` | pkg:pypi/python-dateutil@2.8.2 | HIGH | 7 | Highest | 4

suppression xml fragment we use is

```
    <suppress>
        <notes><![CDATA[
file name: python-dateutil:2.8.2

dev notes: seems to be a false-positive CPE software identification, 
mistaking from python-dateutil:2.8.2 (up-to-date, no vulnerabilities package) that
python:2.8.2 (old core python version) is in use, which is flagging multiple
CVE vulnerabilities.
      ]]>        </notes>
        <packageUrl regex="true">^pkg:pypi/python\-dateutil@.*$</packageUrl>
        <cpe>cpe:/a:python:python:2.8.2</cpe>
        <cpe>cpe:/a:python_software_foundation:python:2.8.2</cpe>
    </suppress>
```

**Comment by aikebah** [comment]
Supposedly resolved since 6.3.0 by https://github.com/jeremylong/DependencyCheck/commit/e1e3bf588e24c360de85bac6bcf045c05a049b3c
